vscode-extension/src/aiConfigEditor.ts

```diff
@@ -144,23 +144,58 @@ export class AIConfigEditorProvider implements vscode.CustomTextEditorProvider {
             return;
           }
 
+          // TODO: Should we skip events with no content changes?
+          // if (e.contentChanges.length === 0) {
+          //   console.log(
+          //     `changeDocumentSubscription ${e.document.uri} - skipping event because there are no content changes`
+          //   );
+          //   return;
+          // }
+
           // TODO: saqadri - instead of sending the entire document to the webview,
           // can ask it to reload the document from the server
 
           // Update webview immediately, and server state should be updated in the background.
+          console.log(
+            `changeDocumentSubscription ${e.document.uri} -- updating webview`
+          );
           updateWebview();
 
           // Notify server of updated document
           if (editorServer) {
-            console.log("changeDocumentSubscription -- updating server");
-
             // TODO: saqadri - decide if we want to await here or just fire and forget
-            await initializeServerState(editorServer.url, e.document);
-          }
+            try {
+              console.log("changeDocumentSubscription -- updating server");
+              await updateServerState(editorServer.url, e.document);
+            } catch (e) {
+              if (isWebviewDisposed || !editorServer) {
+                // Ignore errors caused by closing the webview while the server update
+                // request is in flight (e.g. closing config w/ unsaved changes will
+                // emit onDidChangeTextDocument with reverted unsaved changes)

```

Comment:
Nit: add something like "please refer to https://github.com/lastmile-ai/aiconfig/pull/1201 for full details" becuase the explanation you have in summary is really helpful